ironfish-rust/src/transaction/tests.rs

```diff
@@ -18,14 +18,22 @@ use ironfish_zkp::redjubjub::Signature;
 fn test_transaction() {
     let spender_key = SaplingKey::generate_key();
     let receiver_key = SaplingKey::generate_key();
+    let sender_key = SaplingKey::generate_key();
```

Comment:
Why do we need 3 keys, don't we just need receiver and sender?

Reply:
there are two notes here, in_note and out_note.
In_note: 
- owner: spender
- sender: sender

Out_note:
- owner: receiver
- sender: spender
cc @rohanjadvani @mat-if

Reply:
Ahh I see, in/out is referencing with relationship to spender.